src/app.js

```diff
@@ -1,48 +1,47 @@
-export default function (options) {
-	var model = options.model
-	var view = options.view
+export default function (app) {
+	var view = app.view || function () {
+		return ""
+	}
 
+	var model
 	var actions = {}
-	var effects = options.effects || {}
-	var reducers = options.reducers || {}
-	var subs = options.subscriptions
-
-	var router = options.router
-
+	var reducers
+	var effects
+	var subscriptions = []
+	var hooks = []
+	var plugins = app.plugins
 	var node
 	var root
 	var batch = []
 
-	var hooks = merge({
-		onAction: Function.prototype,
-		onUpdate: Function.prototype,
-		onError: function (err) {
+	var onError = function (err) {
+		if (!hook("onError", function (fn) {
```

Comment:
Maybe I'm missing something, but I still think this is not going to work when sharing onError with plugins. This is a minor thing, so please send me a PR after we land this one 👍.